addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/CsrfCountermeasuresScanRule.java

```diff
@@ -247,6 +249,16 @@ private boolean formOnIgnoreList(Element formElement, List<String> ignoreList) {
         return false;
     }
 
+    private static BiPredicate<String, String> getMatcher() {
+        if (Model.getSingleton()
+                .getOptionsParam()
+                .getParamSet(AntiCsrfParam.class)
+                .isPartialMatchingEnabled()) {
+            return StringUtils::containsIgnoreCase;
+        }
+        return String::equalsIgnoreCase;
+    }
+
```

Comment:
Should this be in a static block at the top? Or do passive scan rules have something like an init method? (Side note: if not, should they?)